monitoring/monitoring.go

```diff
@@ -63,6 +63,10 @@ type (
 		// Register configures the Metric for export. It MUST be called before collection
 		// of values for the Metric. An error will be returned if registration fails.
 		Register() error
+
+		// disable configures the metric in 'Disabled' state. It is useful for cases where
+		// metrics are enabled based on some configuration flag.
+		disable()
```

Comment:
TIL interfaces can have private methods. Makes sense, I had just never seen it before.